kld/src/database/mod.rs

```diff
@@ -231,3 +233,42 @@ pub fn microsecond_timestamp() -> OffsetDateTime {
 pub fn to_primative(offset: &OffsetDateTime) -> PrimitiveDateTime {
     PrimitiveDateTime::new(offset.date(), offset.time())
 }
+
+pub trait RowExt {
+    fn get_u64(&self, column: &str) -> u64;
+    fn get_timestamp(&self, column: &str) -> OffsetDateTime;
+    fn get_timestamp_optional(&self, column: &str) -> Option<OffsetDateTime>;
+    fn maybe_read<T: MaybeReadable>(&self, column: &str) -> Result<T>;
+    fn maybe_read_optional<T: MaybeReadable>(&self, column: &str) -> Result<Option<T>>;
+}
+
+impl RowExt for Row {
+    fn get_u64(&self, column: &str) -> u64 {
+        self.get::<&str, i64>(column) as u64
+    }
+
+    fn maybe_read<T: MaybeReadable>(&self, column: &str) -> Result<T> {
+        T::read(&mut self.get::<&str, &[u8]>(column))
+            .map_err(decode_error)?
+            .context(format!("expected readable value for column {column}"))
+    }
+
+    fn maybe_read_optional<T: MaybeReadable>(&self, column: &str) -> Result<Option<T>> {
+        self.get::<&str, Option<&[u8]>>(column)
+            .map(|mut bytes| {
+                T::read(&mut bytes)
+                    .map_err(decode_error)?
+                    .context(format!("expected readable value for column {column}"))
+            })
+            .transpose()
+    }
+
+    fn get_timestamp(&self, column: &str) -> OffsetDateTime {
+        self.get::<&str, PrimitiveDateTime>(column).assume_utc()
+    }
+
+    fn get_timestamp_optional(&self, column: &str) -> Option<OffsetDateTime> {
+        self.get::<&str, Option<PrimitiveDateTime>>(column)
+            .map(|time| time.assume_utc())
+    }
+}
```

Comment:
Ok I can see how get_u64 might be misused/misleading so I've removed that.

I added the maybe_ prefix because these functions accept MaybeReadable types as opposed to Readable type. But I see that MaybeReadable implements Readable so I think we can remove the prefix for simplicity. 
I think I still need the optional function though. Because database columns can be nullable or not. If a non nullable column is null then that's an error. I need to be able to explicitly say a column is optional or not.